Bluepill-cli/Bluepill-cli/Simulator/BPSimulator.m

```diff
@@ -294,7 +294,7 @@ - (void)deleteSimulatorWithCompletion:(void (^)(NSError *error, BOOL success))co
             }
             completion(error, error ? NO: YES);
         }];
-    } else {
+    } else if (self.device) {
```

Comment:
@vargon I moved the simulator check from https://github.com/linkedin/bluepill/pull/131/files#diff-1070fc70916a7ce09ff5de4c27a7129bL470 to here because if this method was called and the device was nil, bluepill would hang forever, so I thought this was a safer place to put it. If you want I can add back an `if (!context.runner.device)` check where it was as well so we don't waste cycles creating the SimServiceContext and SimDeviceSet.